build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -57,11 +82,26 @@ def main(args: argparse.Namespace):
 
   target_device_names = (set(args.target_device_names)
                          if args.target_device_names is not None else None)
+  device_spec_matchers = None
+  if args.benchmark_presets is not None:
+    device_spec_matchers = []
+    for preset in args.benchmark_presets:
+      matcher = BENCHMARK_PRESET_DEVICE_SPEC_MATCHERS.get(preset)
+      if matcher is None:
+        raise ValueError(f"Unrecognized benchmark preset: '{preset}'.")
```

Comment:
Should this be `ArgumentError`? Also probably nice to include what the valid options are here and in the argument help text

Reply:
I move this into the part of argument parsing (https://docs.python.org/3/library/argparse.html#type). ArgumentParser should handle the error message better.

Reply:
Matcher also knows takes the whole E2EModelRunConfig, which looks more general to me.